### PR TITLE
Support pod annotations in helm chart values

### PR DESCRIPTION
**Pull request by avivkarbolk**
Support annotations values under pod scope

### CODE REVIEW

File: helm/values.yaml [hunk 1]
```diff
@@ -25,6 +25,7 @@ coder:
   # coder.annotations -- The Deployment annotations. See:
   # https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/
   annotations: {}
```
Comment (kylecarbs) [inline helm/values.yaml:27]: We should probably rename `annotations` to `deployment_annotations` then and deprecate `annotations`.

If you aren't comfortable with doing that just let us know and we'll take it over!

**Review by kylecarbs — COMMENTED** [review]
Thanks for the contribution! 🥳

File: helm/values.yaml (hunk 1 above)
Comment (avivkarbolk) [reply helm/values.yaml:27]: @kylecarbs Hi, I'm happy to contribute. I will rename it :)

Comment (avivkarbolk) [reply helm/values.yaml]: @kylecarbs wdyt?

**Pull request closed without merging**